### PR TITLE
Volume.getMigrateStatus should return null by default

### LINKED ISSUE #9: Setting Volume name and description fails with Cinder "Rocky"
[BlockVolumeService.update(String volumeId, String name, String description)](https://github.com/ContainX/openstack4j/blob/3.2.0/core/src/main/java/org/openstack4j/openstack/storage/block/internal/BlockVolumeServiceImpl.java#L138) is used to set the name and description on a (Cinder) Volume.
This results in a JSON request being HTTP PUT to OpenStack containing the new name & description ... but it also includes `os-vol-mig-status-attr:migstat` = `none` with the rest of the JSON request, and the "Rocky" version of Cinder rejects that request:
```
ActionResponse{success=false, fault=Invalid input for field/attribute volume. Value: {u'description': u'...', u'display_name': u'...', u'name': u'...', u'os-vol-mig-status-attr:migstat': u'none', u'display_description': u'...'}. Additional properties are not allowed (u'os-vol-mig-status-attr:migstat' was unexpected), code=400}
```

i.e. the `update` method sets the `name` & `display_name` attributes to the `String name` it was given, and both the `description` & `display_description` attributes to the `String description` it was given, but it also tries to set attribute `os-vol-mig-status-attr:migstat` to `none`, which fails.

From what I've seen thus far, the underlying cause appears to be that [CinderVolume.getMigrateStatus()](https://github.com/openstack4j/openstack4j/blob/openstack4j-parent-3.3/core/src/main/java/org/openstack4j/openstack/storage/block/domain/CinderVolume.java#L182) does not return null if it has not been set; this then causes the JSON code to mistakenly believe it should be included in the outgoing request ... which causes Cinder to reject the (invalid) request.
According to the [javadocs](https://github.com/openstack4j/openstack4j/blob/openstack4j-parent-3.3/core/src/main/java/org/openstack4j/model/storage/block/Volume.java#L164), this field's default value should be null, but the existing implementation returns a default value of `MigrationStatus.NONE` instead of null.

i.e. it's not working as documented & it isn't working; this is probably cause & effect.

Note: The "Pike" version of Cinder doesn't seem to care about this and does not reject the request, allowing this issue to remain undetected for so long; the "Rocky" version does care and does reject it; I don't know about the "Queen" version.
Note also: The openstack4j has had numerous similar bugs in the past.  e.g. ContainX/openstack4j [#932](https://github.com/ContainX/openstack4j/pull/932), [#869](https://github.com/ContainX/openstack4j/issues/869), [#823](https://github.com/ContainX/openstack4j/issues/823), [#820](https://github.com/ContainX/openstack4j/pull/820), [#682](https://github.com/ContainX/openstack4j/issues/682), [#606](https://github.com/ContainX/openstack4j/issues/606), [#573](https://github.com/ContainX/openstack4j/issues/573), [#470](https://github.com/ContainX/openstack4j/issues/470).

### PR DESCRIPTION
**Pull request by pjdarton**
Fix #9 
...and hopefully fix [Jenkins OpenStack-cloud-plugin bug 277](https://github.com/jenkinsci/openstack-cloud-plugin/issues/277) in the process.

Summary of changes:

1. The CinderVolume POJO now "just does what it's told" and doesn't second-guess what the MigrationStatus should be.
This should then fix the issue where the outgoing JSON request updated the migration status when no such change was requested.

1. Any (external) caller _reading_ this field should be sufficiently null-proofed by Volume.MigrationStatus.fromValue's null-handling functionality where, when turning JSON to Java, a null JSON string is turned into MigrationStatus.NONE.

i.e. the POJO will (now) retain null as null when writing but anyone reading the data back again will read MigrationStatus.NONE.

### CODE REVIEW

**Comment by pjdarton** [comment]
I'm no expert on Travis and I struggle to understand exactly what the failure it reported concerns ... but it built fine on my local machine; I'm not convinced that the CI build failure on Travis is a direct result of my changes...

**Comment by olivergondza** [comment]
@pjdarton,  welcome to openstack4j/openstack4j!

Please, merge the master. I have fixed the travis build a minute ago.

**Comment by pjdarton** [comment]
I've not tested it "for real" ... but if you can tell me (or point me at instructions) how to do a build of the openstack Jenkins plugin that uses a custom build of openstack4j then I can test it "for real" like that.

**Comment by olivergondza** [comment]
What I guess would be enough is to make sure openstack4j part is working. I use [tester groovy script](https://gist.github.com/olivergondza/2fd7656a57ef485af133e6de3bef6d87) like so:
```
$ mvn clean install -DskipTests
$ source openrc.sh
$ groovy test.groovy
```

At the end of the script you add the code verifying the interaction that used to fail now succeeds. (Given the coverage levels, this might be good contribution guideline for future RPs).

--

Re building openstack-cloud plugin: installing openstack4j locally and then bumping the dependency (to the snapshot one) in the plugin's pom it really all it takes.

**Comment by pjdarton** [comment]
I've tested it by embedding my code-changed-openstack4j into the Jenkins openstack plugin (which is where I originally encountered [this bug](https://github.com/jenkinsci/openstack-cloud-plugin/issues/277)).
That code sets the name and description on the Volume used by the Instance it's created ... except that fails without this fix.

Without this PR (but with Jenkins openstack-cloud-plugin [PR278](https://github.com/jenkinsci/openstack-cloud-plugin/pull/278) to make failure non-fatal), the running instance shows (in the Horizon WebUI):
image
and that volume shows:
image
and the exception described in Jenkins openstack-cloud-plugin [issue 277](https://github.com/jenkinsci/openstack-cloud-plugin/issues/277) is logged.

With this PR, the running instance shows:
image
and that volume shows:
image
and no such exception is encountered.

i.e. It's now possible to set the name and description in a Cinder volume.

TL;DR: I think it works.

**Comment by pjdarton** [comment]
I'm going to close, wait a bit, then re-open the PR in the hope that this will re-trigger the Travis CI build.

**Comment by pjdarton** [comment]
I've rebased on master, the CI build is clean, I've manually tested it and it seems good.
It's ready for merge :+1: 

Given your earlier stated approval, I would do that myself but github tells me "Merging is blocked" & "You’re not authorized to merge this pull request." ... so I guess merely being a member of https://github.com/openstack4j doesn't automatically include commit access :wink:

**Comment by olivergondza** [comment]
@pjdarton, Thanks. Merging right away.

Re maintainership, you are first on the list to be asked to become a co-maintainer. Let me put together a doc of what does it entails and I will get back to you with a formulated procedure.
